app/src/main/java/eu/faircode/email/HtmlHelper.java

```diff
@@ -324,7 +324,10 @@ static Drawable decodeImage(String source, Context context, long id, boolean sho
 
     static String getPreview(String body) {
         String text = (body == null ? null : Jsoup.parse(body).text());
-        return (text == null ? null : text.substring(0, Math.min(text.length(), PREVIEW_SIZE)));
+        if (text == null || text.length() <= PREVIEW_SIZE) {
+            return text;
+        }
+        return text.substring(0, PREVIEW_SIZE) + '\u2026';
```

Comment:
There is no need for this, since the layout is set to ellipsize already:

https://github.com/M66B/open-source-email/blob/master/app/src/main/res/layout/item_message_normal.xml#L250

Reply:
This is intended for the preview in expanded notifications where it is just cut off after `PREVIEW_SIZE` characters and it doesn't hurt the preview in TextView you linked.

Reply:
I have solved it like this because I don't think what goes into the database should be changed: https://github.com/M66B/open-source-email/commit/1ed0fe7fce07b25f6bf31234d7ca0cd46c59e19a